### PR TITLE
RDM-6472 Functional tests for get jurisdictions

### PR DESCRIPTION
**Pull request by RebeccaBaker**
# JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-647

### Change description ###
Functional tests for get jurisdictions

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: aat/src/aat/resources/features/F-081/S-347.td.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      "Content-Length" : "[[ANYTHING_PRESENT]]",
+      "Content-Type" : "application/json;charset=UTF-8",
+      "Date" : "[[ANYTHING_PRESENT]]",
+      "Expires" : "0",
+      "Pragma" : "no-cache",
+      "Vary" : "accept-encoding",
+      "X-Content-Type-Options" : "nosniff",
+      "X-Frame-Options" : "DENY",
+      "X-Xss-Protection" : "1; mode=block"
+    },
```
Comment (MSancaktutar) [inline aat/src/aat/resources/features/F-081/S-347.td.json]: these lines could mostly be inherited from a common object.

File: aat/src/aat/resources/features/F-081/F-081.feature [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  Background:
+    Given an appropriate test context as detailed in the test data source
+
+  @S-348 @Ignore # Response code mismatch, expected: 401, actual: 403 RDM-6628
+  Scenario: must return 401 when request does not provide valid authentication credentials
+    Given a user with [an active profile in CCD]
+    When a request is prepared with appropriate values
+    And the request [contains an invalid authentication credentials]
+    And it is submitted to call the [get jurisdiction details] operation of [CCD Definition Store]
```
Comment (MSancaktutar) [inline aat/src/aat/resources/features/F-081/F-081.feature]: Get Jurisdiction Details (first letters upper case to denote a name/title)

File: aat/src/aat/resources/features/F-081/F-081.feature [hunk 3]
```diff
@@ -0,0 +1,43 @@
+@F-081
+Feature: CCD Definition Store Api :: GET /api/data/jurisdictions
```
Comment (MSancaktutar) [inline aat/src/aat/resources/features/F-081/F-081.feature]: Please use the same name you suggested below, for consistency - Get Jurisdiction Details

File: aat/src/aat/resources/features/F-081/F-081.feature [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  Scenario: must return 403 when request provides authentic credentials without authorised access to the operation
+    Given a user with [an active profile in CCD, and insufficient privilege to the case type]
+    When a request is prepared with appropriate values
+    And it is submitted to call the [get jurisdiction details] operation of [CCD Definition Store]
+    Then a negative response is received
+    And the response [contains 403 forbidden code]
+    And the response has all the details as expected
+
+  @S-347 #RDM-6858
+  Scenario: must return 200 with List of jurisdictions
```
Comment (MSancaktutar) [inline aat/src/aat/resources/features/F-081/F-081.feature]: happy path scenarios always on top in the definition file, please.

File: aat/src/aat/resources/features/F-081/F-081.feature [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    Given a user with [an active profile in CCD]
+    When a request is prepared with appropriate values
+    And the request [contains a valid jurisdiction id]
+    And it is submitted to call the [get jurisdiction details] operation of [CCD Definition Store]
+    And the response [has the 200 OK code]
```
Comment (MSancaktutar) [inline aat/src/aat/resources/features/F-081/F-081.feature]: better also mentioning what should be returned in the response body.

File: aat/src/aat/resources/features/F-081/S-350.td.json [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      "Content-Encoding" : "gzip",
+      "Content-Length" : "85",
+      "Content-Type" : "application/json;charset=UTF-8",
+      "Date" : "[[ANYTHING_PRESENT]]",
+      "Expires" : "0",
+      "Pragma" : "no-cache",
+      "Vary" : "accept-encoding",
+      "X-Content-Type-Options" : "nosniff",
+      "X-Frame-Options" : "DENY",
+      "X-Xss-Protection" : "1; mode=block"
```
Comment (MSancaktutar) [inline aat/src/aat/resources/features/F-081/S-350.td.json]: these lines should all or mostly be inherited from a common object.

File: aat/src/aat/resources/features/F-081/F-081.feature [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    Given an appropriate test context as detailed in the test data source
+
+  @S-347 #RDM-6858
+  Scenario: must return 200 with List of jurisdictions
+    Given a user with [an active profile in CCD]
+    When a request is prepared with appropriate values
+    And the request [contains a valid jurisdiction id]
+    And it is submitted to call the [Get Jurisdiction Details] operation of [CCD Definition Store]
+    And the response [has the 200 OK code]
+    And the response [contains the jurisdictions details for jurisdiction id BEFTA_CASETYPE_2_1]
```
Comment (MSancaktutar) [inline aat/src/aat/resources/features/F-081/F-081.feature]: This wording is a specific reference to a jurisdiction. The functional spec by a BA, on the other hand, would be in my general terms.

Comment (MSancaktutar) [reply aat/src/aat/resources/features/F-081/F-081.feature]: And the response [contains the jurisdictions details for jurisdiction id provided to the request]

File: aat/src/aat/resources/features/F-081/S-347.td.json [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  "expectedResponse": {
+    "_extends_" : "Common_200_Response",
+    "headers": {
+      "Content-Encoding" : "gzip",
+      "Content-Length" : "[[ANYTHING_PRESENT]]",
+      "Content-Type" : "application/json;charset=UTF-8",
+      "Strict-Transport-Security": "[[ANY_STRING_NULLABLE]]"
+    },
+    "body" : {
+      "arrayInMap" : "[[ANYTHING_PRESENT]]"
```
Comment (MSancaktutar) [inline aat/src/aat/resources/features/F-081/S-347.td.json]: If we are getting a specific jurisdiction, why are we not specifying its details inside the array??

Comment (RebeccaBaker) [reply aat/src/aat/resources/features/F-081/S-347.td.json]: the response does not have a consistent ordering to enable us to have a more defined expected response

Comment (RebeccaBaker) [reply aat/src/aat/resources/features/F-081/S-347.td.json]: hence i have linked to a ticket to rectify in the feature file

Comment (MSancaktutar) [reply aat/src/aat/resources/features/F-081/S-347.td.json]: but this one is returning only 1 jurisdiction object in the array, isn't it?

Comment (RebeccaBaker) [reply aat/src/aat/resources/features/F-081/S-347.td.json]: been able to be more specific with the returned fields, for case type however there is no ordering in which case type is returned first

File: aat/src/aat/resources/features/F-081/F-081.feature [hunk 9]
```diff
@@ -0,0 +1,44 @@
+@F-081
+Feature: CCD Definition Store Api : Get Jurisdiction Details
```
Comment (MSancaktutar) [inline aat/src/aat/resources/features/F-081/F-081.feature]: Feature: F-081: Get Jurisdiction Details

**Pull request merged**